### PR TITLE
[reaper] Add debug mode to save reports

### PR DESCRIPTION
**Pull request by chromy**

### CODE REVIEW

File: reaper/reaper/src/main/kotlin/com/emergetools/reaper/ReaperReportUploadWorker.kt [hunk 1]
```diff
@@ -207,6 +228,6 @@ fun longToBase64(hash: Long): String {
   for (i in 0..7) {
     buf[i] = ((hash shr i * 8) and 0xFFL).toByte()
   }
-  val hashAsBase64 = Base64.encodeToString(buf, Base64.DEFAULT)
-  return hashAsBase64
+  val hashAsBase64 = Base64.encodeToString(buf, Base64.NO_WRAP)
```
Comment (rbro112) [inline reaper/reaper/src/main/kotlin/com/emergetools/reaper/ReaperReportUploadWorker.kt:231]: I think you'll need to rebase onto https://github.com/EmergeTools/emerge-android/pull/214/files

File: reaper/reaper/src/main/kotlin/com/emergetools/reaper/ReaperInternal.kt [hunk 2]
```diff
@@ -128,6 +129,11 @@ internal object ReaperInternal {
       fatalError(context, "Manifest com.emergetools.PUBLISHABLE_API_KEY must be set and non-empty.")
     }
 
+    isDebug = context.packageManager.getApplicationInfo(
+      context.packageName,
+      PackageManager.GET_META_DATA
+    ).metaData.getBoolean(MANIFEST_TAG_DEBUG, false)
```
Comment (rbro112) [inline reaper/reaper/src/main/kotlin/com/emergetools/reaper/ReaperInternal.kt:135]: What's the plan here, assuming there will be a grade plugin tie-in to set this tag from an extension property?

**Review by rbro112 — APPROVED** [review]

File: reaper/reaper/src/main/kotlin/com/emergetools/reaper/ReaperReportUploadWorker.kt (hunk 1 above)
Comment (chromy) [reply reaper/reaper/src/main/kotlin/com/emergetools/reaper/ReaperReportUploadWorker.kt]: thanks, done!

File: reaper/reaper/src/main/kotlin/com/emergetools/reaper/ReaperInternal.kt (hunk 2 above)
Comment (chromy) [reply reaper/reaper/src/main/kotlin/com/emergetools/reaper/ReaperInternal.kt:135]: So far I've been manually adding:
```
<meta-data
   android:name="com.emergetools.reaper.DEBUG"
   android:value="true" />
```
to the manifest when needed. Would be neat to hook it up to the grade plugin though. I think the plugin already exposes a 'debug' setting iirc

**Pull request merged**